examples/plot_metadata_routing.py

```diff
@@ -0,0 +1,612 @@
+        # and then prepare the input to the underlying ``predict`` method.
+        params = request_router.route_params(params=predict_params, caller="predict")
+        return self.estimator_.predict(X, **params.estimator.predict)
+
+    def get_metadata_routing(self):
+        router = (
+            MetadataRouter(owner=self.__class__.__name__)
+            .add_self(self)
+            .add(estimator=self.estimator, method_mapping="one-to-one")
+        )
+        return router
+
+
+# %%
+# The key parts where the above estimator differs from our previous
+# meta-estimator is accepting ``sample_weight`` explicitly in ``fit`` and
+# including it in ``fit_params``. Making ``sample_weight`` an explicit argument
+# makes sure ``fit_requests(sample_weight=...)`` is present for this class.
+#
+# In ``get_metadata_routing``, we add ``self`` to the routing using
+# ``add_self``. Now let's look at some examples:
+
+# %%
+# no metadata requested
+est = RouterConsumerClassifier(estimator=ExampleClassifier())
+print_routing(est)
+
+
+# %%
+# ``sample_weight`` requested by child estimator
+est = RouterConsumerClassifier(
+    estimator=ExampleClassifier().fit_requests(sample_weight=True)
+)
+print_routing(est)
+
+# %%
+# ``sample_weight`` requested by meta-estimator
+est = RouterConsumerClassifier(estimator=ExampleClassifier()).fit_requests(
+    sample_weight=True
+)
+print_routing(est)
+
+# %%
+# Note the difference in the requested meatada representations above.
+#
+# We can also alias the metadata to pass different values to them:
+
+est = RouterConsumerClassifier(
+    estimator=ExampleClassifier().fit_requests(sample_weight="clf_sample_weight"),
+).fit_requests(sample_weight="meta_clf_sample_weight")
+print_routing(est)
+
+# %%
+# However, ``fit`` of the meta-estimator only needs the alias for the
+# sub-estimator, since it doesn't validate and route its own required metadata:
+est.fit(X, y, sample_weight=my_weights, clf_sample_weight=my_other_weights)
+
+# %%
+# Alias only on the sub-estimator. This is useful if we don't want the
+# meta-estimator to use the metadata, and we only want the metadata to be used
+# by the sub-estimator.
+est = RouterConsumerClassifier(
+    estimator=ExampleClassifier().fit_requests(sample_weight="aliased_sample_weight")
+).fit_requests(sample_weight=True)
+print_routing(est)
+
+
+# %%
+# Simple Pipeline
+# ---------------
+# A slightly more complicated use-case is a meta-estimator which does something
+# similar to the ``Pipeline``. Here is a meta-estimator, which accepts a
+# transformer and a classifier, and applies the transformer before running the
+# classifier.
+
+
+class SimplePipeline(ClassifierMixin, BaseEstimator):
+    _required_parameters = ["estimator"]
+
+    def __init__(self, transformer, classifier):
+        self.transformer = transformer
+        self.classifier = classifier
+
+    def fit(self, X, y, **fit_params):
+        params = process_routing(self, "fit", fit_params)
```

Comment:
why don't you use the same type of code here to get params? meaning why not using a router object and a call to request_router.route_params?

Reply:
because that code is a bit more complicated for estimators which are consumers and routers at the same time. If there are args passed to this method, they need to be merged with `fit_params`, and that's extra boilerplate code repeated in meta-estimators:

https://github.com/scikit-learn/scikit-learn/blob/2f03a1bd14cef5ccc290057ebf1daec38b9257b8/sklearn/utils/_metadata_requests.py#L1123-L1124

I wouldn't feel comfortable merging validate and get params since I remember having issues in https://github.com/scikit-learn/scikit-learn/pull/16079 when dealing with deprecations and the usecases we have in scikit-learn.